catris.py

```diff
@@ -457,42 +502,44 @@ async def _bomb_task(self) -> None:
         while True:
             await asyncio.sleep(1)
 
-            bombs: list[tuple[int, int, Bomb, Player | None]] = []
-            for (x, y), value in self.landed_blocks.items():
-                if isinstance(value, Bomb):
-                    bombs.append((x, y, value, None))
-
-            for player in self.players:
-                block = player.moving_block_or_wait_counter
-                if isinstance(block, MovingBlock) and block.bomb is not None:
-                    for x, y in self.get_moving_block_coords(block):
-                        bombs.append((x, y, block.bomb, player))
-
-            # Same Bomb object can appear multiple times. Decrement only once.
-            for bomb in set(b for x, y, b, player in bombs):
-                bomb.timer -= 1
+            bombs: list[BombSquare] = [
+                square
+                for square in self._get_all_squares()
+                if isinstance(square, BombSquare)
+            ]
 
             exploding_points = set()
-            for bomb_x, bomb_y, bomb, player_who_moves_bomb in bombs:
+            for bomb in bombs:
+                bomb.timer -= 1
                 if bomb.timer == 0:
                     print("Bomb explodes! BOOOOOOMM!!!11!")
 
                     radius = 3.5
                     exploding_points |= {
                         (x, y)
-                        for x, y in self.landed_blocks.keys()
-                        if (x - bomb_x) ** 2 + (y - bomb_y) ** 2 < radius**2
+                        for x, y in self.valid_landed_coordinates
+                        if (x - bomb.x) ** 2 + (y - bomb.y) ** 2 < radius**2
                     }
-                    if player_who_moves_bomb is not None:
-                        player_who_moves_bomb.moving_block_or_wait_counter = (
-                            MovingBlock(player_who_moves_bomb)
-                        )
 
             if exploding_points:
                 async with self.flashing_lock:
                     await self.flash(exploding_points, 41)
-                    for point in exploding_points:
-                        self.landed_blocks[point] = None
+                    for square in self.landed_squares.copy():
+                        if (square.x, square.y) in exploding_points:
+                            self.landed_squares.remove(square)
+                    for player in self.players:
+                        if isinstance(player.moving_block_or_wait_counter, MovingBlock):
+                            for (
+                                square
+                            ) in player.moving_block_or_wait_counter.squares.copy():
```

Comment:
too deep indentation + black = lol

Should at least add a new variable `block = player.moving_block_or_wait_counter` or similar.